From e0ffe7dc2af8a2fb2e860b7fa342baa6a5324fd9 Mon Sep 17 00:00:00 2001
From: James Wei <jameswei@google.com>
Date: Fri, 3 Dec 2021 14:00:12 +0800
Subject: [PATCH] Revert "Increase debounce time for DISCONNECT processing"

This reverts commit 5a14ba686170b0475a546df66ad8c7e0c6ee6abc.
(from AOSP)

Bug: 207057578
Test: manual: switch to MTP mode then re-plug

Change-Id: I1e0b33395e164e566df6c8b5007e4cc1d5cd73b9
(cherry picked from commit 54f4c1843d4d41fb784f416575ec8b9857e3d195)
Merged-In:I1e0b33395e164e566df6c8b5007e4cc1d5cd73b9
---
 .../java/com/android/server/usb/UsbDeviceManager.java    | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/services/usb/java/com/android/server/usb/UsbDeviceManager.java b/services/usb/java/com/android/server/usb/UsbDeviceManager.java
index 45b2fce0c4e6..849e6b1615eb 100644
--- a/services/usb/java/com/android/server/usb/UsbDeviceManager.java
+++ b/services/usb/java/com/android/server/usb/UsbDeviceManager.java
@@ -177,10 +177,7 @@ public class UsbDeviceManager implements ActivityTaskManagerInternal.ScreenObser
     // Delay for debouncing USB disconnects.
     // We often get rapid connect/disconnect events when enabling USB functions,
     // which need debouncing.
-    private static final int DEVICE_STATE_UPDATE_DELAY = 3000;
-
-    // Delay for debouncing USB disconnects on Type-C ports in host mode
-    private static final int HOST_STATE_UPDATE_DELAY = 1000;
+    private static final int UPDATE_DELAY = 1000;
 
     // Timeout for entering USB request mode.
     // Request is cancelled if host does not configure device within 10 seconds.
@@ -650,7 +647,7 @@ public class UsbDeviceManager implements ActivityTaskManagerInternal.ScreenObser
             msg.arg1 = connected;
             msg.arg2 = configured;
             // debounce disconnects to avoid problems bringing up USB tethering
-            sendMessageDelayed(msg, (connected == 0) ? DEVICE_STATE_UPDATE_DELAY : 0);
+            sendMessageDelayed(msg, (connected == 0) ? UPDATE_DELAY : 0);
         }
 
         public void updateHostState(UsbPort port, UsbPortStatus status) {
@@ -665,7 +662,7 @@ public class UsbDeviceManager implements ActivityTaskManagerInternal.ScreenObser
             removeMessages(MSG_UPDATE_PORT_STATE);
             Message msg = obtainMessage(MSG_UPDATE_PORT_STATE, args);
             // debounce rapid transitions of connect/disconnect on type-c ports
-            sendMessageDelayed(msg, HOST_STATE_UPDATE_DELAY);
+            sendMessageDelayed(msg, UPDATE_DELAY);
         }
 
         private void setAdbEnabled(boolean enable) {
-- 
2.33.1

